descartes_light/solvers/include/descartes_light/solvers/bgl/impl/event_visitors.hpp

```diff
@@ -32,6 +32,75 @@ struct early_terminator : public boost::base_visitor<early_terminator<FloatType>
   const long last_rung_idx_;
 };
 
+/**
+ * @brief Event visitor that adds all edges to each vertex dynamically as the vertex is discovered during the graph
+ * search
+ */
+template <typename FloatType>
+struct add_all_edges_dynamically : public boost::base_visitor<add_all_edges_dynamically<FloatType>>
+{
+  /** @brief Event filter typedef defining the events for which this visitor can be used */
+  typedef boost::on_discover_vertex event_filter;
+
+  add_all_edges_dynamically(std::vector<typename EdgeEvaluator<FloatType>::ConstPtr>& edge_eval,
+                            std::vector<std::vector<VertexDesc<FloatType>>>& ladder_rungs)
+    : eval_(edge_eval), ladder_rungs_(ladder_rungs)
+  {
+  }
+
+  void operator()(VertexDesc<FloatType> u, const BGLGraph<FloatType>& g)
+  {
+    auto out_deg = boost::out_degree(u, g);
+    // return if the vertex has any out edges
+    if (out_deg == 0)
+    {
+      std::size_t current_rung = static_cast<std::size_t>(g[u].rung_idx);
+      std::size_t next_rung = static_cast<std::size_t>(current_rung + 1);
+      if (next_rung < ladder_rungs_.size())
+      {
+        FloatType cost;
+        for (std::size_t s = 0; s < ladder_rungs_[next_rung].size(); ++s)
+        {
+          std::pair<bool, FloatType> results = eval_[static_cast<size_t>(current_rung)]->evaluate(
+              *g[u].sample.state, *g[ladder_rungs_[next_rung][s]].sample.state);
+          if (results.first)
+          {
+            cost = results.second + g[ladder_rungs_[next_rung][s]].sample.cost;
```

Comment:
```suggestion
            FloatType cost = results.second + g[ladder_rungs_[next_rung][s]].sample.cost;
```

Declare the variable where you use it (rather than above) for clarity